src/nvim/CMakeLists.txt

```diff
@@ -130,6 +130,12 @@ glob_wrapper(UNIT_TEST_FIXTURES ${PROJECT_SOURCE_DIR}/test/unit/fixtures/*.c)
 list(SORT NVIM_SOURCES)
 list(SORT NVIM_HEADERS)
 
+add_library(nvim_sources INTERFACE)
+target_sources(nvim_sources INTERFACE ${NVIM_SOURCES})
+
+add_library(nvim_headers INTERFACE ${NVIM_HEADERS})
+target_sources(nvim_headers INTERFACE ${NVIM_HEADERS})
```

Comment:
if this is truly needed, it should be coupled with removing the other code where we are collecting sources/headers and building nvim. We should not have 2 different places where we're doing the same thing in a different way.

Reply:
This is done immediately after these collections are created. This list is currently only used in `src/nvim/CMakeLists.txt` and its _child_ directory `src/nvim/po/CMakeLists.txt`, so unless I missed something, this is the first time that this was needed.

That said, we can use this same "modern" pattern to replace this and other similar usages of directory-based properties with target-based ones
https://github.com/neovim/neovim/blob/3b23f561aef87687495ca683f12943d0cd8eff15/CMakeLists.txt#L666